### PR TITLE
fluxcd: init at 0.5.0

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change

Add [fluxcd](https://fluxcd.io/)

This is flux v2. There is also [fluxctl](https://github.com/NixOS/nixpkgs/blob/dfe13f9152d67dee707c1ac67dbd4c7197ce470e/pkgs/applications/networking/cluster/fluxctl/default.nix) (flux v1). v1 is in maintenance
mode, and it will be deprecated at some point, but some people may still
depend on v1.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  vendorSha256 = "0f818a0z71nl061db93iqb87njx66vbrra1zh92warbx8djdsr7k";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  doCheck = false;
+
+  subPackages = [ "cmd/flux" ];
+
+  buildFlagsArray = [ "-ldflags=-s -w -X main.VERSION=${version}" ];
```
Comment (glittershark) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:22]: I'm not super familiar with the build toolchain, why is this necessary?

**Comment by glittershark** [comment]
Result of `nixpkgs-review pr 106483` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fluxcd</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/fluxcd/default.nix (hunk 1 above)
Comment (glittershark) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:22]: at least the `-ldflags=-s -w` bit, the version bit makes sense

Comment (jlesquembre) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:22]: Thanks for reviewing! 
Both options are necessary, to know more about the go toolchain, you can take a look here: https://nixos.org/manual/nixpkgs/stable/#ssec-go-modules. It explains the `subPackages` option.

The `buildFlagsArray` values are taken from upstream: 

https://github.com/fluxcd/flux2/blob/0ba6fc1b366bd6f2ce231d3ac8a95eac57ff8554/.goreleaser.yml#L6

The `-X main.VERSION` is needed because without it, the `flux --version` command will return version 0

Those two flags are pretty common between the go derivations

Comment (glittershark) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:22]: Thanks for the explanation!!

Comment (timokau) [reply pkgs/applications/networking/cluster/fluxcd/default.nix:22]: Ideally the link to the upstream source would be mentioned in a comment to avoid future confusion. The rest of the diff looks good though, and since its a very common flag this is not very important. Just mentioning this for future PRs. If something needs explanation in the review, its usually best to explain it in a comment. [This](https://mtlynch.io/code-review-love/#4-answer-questions-with-the-code-itself) blog post argues the point very well.

**Review by timokau — APPROVED** [review]
Thanks to both of you :)
